### PR TITLE
78 product subcategory template

### LINKED ISSUE #36: FR: Subcategory Template: Product listing with Card Dynamic Data

### LINKED ISSUE #78: FR: Product Subcategory Template (dynamically created content)
**Description**
The user uses the navigation (categories > electrical > batteries) to arrive to the Batteries Subcategory Template.

**Current situation:**
The structure of the URL to the product subcategory page is formed of:
- Fixed part: https://roadchoice.com/part-category/
- Subcategory: appended to each subcategory page
_Example: https://roadchoice.com/part-category/batteries_

**Future situation:**
The structure of the URL to the subcategory page is formed of:
- Fixed part: https://roadchoice.com/part-category/
- Category (ex. Electrical)
- Subcategory (ex. Batteries)
The data for the URL is retrieved automatically from the subcategory file ([rc_batteries](https://adobe.sharepoint.com/:x:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BD98F331A-BB95-4DA4-8A9F-4BF6021BFF20%7D&file=RC_Batteries.xlsx&action=default&mobileredirect=true&cid=a7c0139f-0568-4726-8439-67791930e0ac)).
_Example: https://roadchoice.com/part-category/electrical/batteries/_
The editor creates and maintains the links to each of the existing subcategories in the Categories subnavigation menu document.

**Data mapping:**
**Data requirement is to find a naming convention for the [excel file](https://adobe.sharepoint.com/:x:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BD98F331A-BB95-4DA4-8A9F-4BF6021BFF20%7D&file=RC_Batteries.xlsx&action=default&mobileredirect=true) to include the subcategory name (proposal: _rc_batteries_**)

**Acceptance criteria**
- The URL follows the described pattern
- The title displayed on the page is according to the subcategory the visitor is seeing
- The product list #36 , filters #37  and pagination #38  are OoS in this ticket, and will be done in their separate tickets 
- Create redirects from the previous URL to the new URL structure

### PR DESCRIPTION
**Pull request by jonatan-lledo-netcentric**
except for those cases where the `rc-[subcategory].json` file is not previewed, or the 2 edge cases of _bumpers_ and _brake-chambers_
the subcategory page loads dynamically based on the parameter in the URL "category"
Tested using _batteries_ and _antennas_ files

also, it handles a 404 page in case something goes wrong

test URLs:
part-category/?category=antennas
part-category/?category=batteries

404 pages:
part-category/
part-category/?
part-category/?cate
part-category/?category
part-category/?category=
part-category/?category=antenn

Fix #78, also fix #36, #37 & #38

Test URLs:
- Before: https://main--vg-roadchoice-com--hlxsites.hlx.page/
- After: https://78-product-subcategory-template--vg-roadchoice-com--hlxsites.hlx.page/

### CODE REVIEW

File: blocks/results-list/product-card.js [hunk 1]
```diff
@@ -45,7 +45,9 @@ const productCard = (product, searchType) => {
   const repository = 'https://adobe.sharepoint.com/:i:/r/sites/HelixProjects/Shared%20Documents/sites/VolvoGroup/vg-roadchoice-com';
 
   // TODO check how the link content should finally be
-  const linkUrl = `${window.location.href}/${category}/${partNumber}`;
+  const windowUrl = new URL(window.location.href);
+  const baseUrl = windowUrl.toString().replace(/\/$/, '');
+  const linkUrl = `${baseUrl}/${category}/${partNumber}`;
```
Comment (manuel-vara) [inline blocks/results-list/product-card.js:50]: To align with PDP page the link should be like this example:  /parts?category=batteries&sku=27FCRC

File: blocks/results-list/product-card.js [hunk 2]
```diff
@@ -45,7 +45,9 @@ const productCard = (product, searchType) => {
   const repository = 'https://adobe.sharepoint.com/:i:/r/sites/HelixProjects/Shared%20Documents/sites/VolvoGroup/vg-roadchoice-com';
```
Comment (manuel-vara) [inline blocks/results-list/product-card.js:45]: Path to Franklin hosted images should be like this example: https://main--vg-roadchoice-com--hlxsites.hlx.page/media/images/media_132095a2f3e1309f212937f07494cdacb7c1b9fb5.jpeg?width=2000&format=webply&optimize=medium

If image is not found then it should be going to look into existing images repo: `parts.trucklibrary.com` , example: https://parts.trucklibrary.com/image/VolvoGroupNorthAmericaLLC/55768-27CRC.0?$s7product$?width=2000&format=webply&optimize=medium 

You can have a look at how images are handled currently in PDP here: 
https://github.com/hlxsites/vg-roadchoice-com/pull/118

**Review by manuel-vara — CHANGES_REQUESTED** [review]

File: blocks/results-list/product-card.js (hunk 2 above)
Comment (Lakshmishri) [reply blocks/results-list/product-card.js:45]: Like said by manuel, fetch from json file where you have to filter by part number, which is found in pdp file under function` fetchPartImages`

File: templates/part-category/part-category.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ * 5. "/part-category/?category=asdf" => 404 if is a wrong category
+*/
+
+/**
+ * Returns the category name from the URL query string, or null if it is not present.
+ * @returns {string|null} The category name, or _null_ if it is not present.
+ */
+const getCategory = async () => {
+  const hasCategory = urlParams.has('category');
+  if (hasCategory) {
```
Comment (Lakshmishri) [inline templates/part-category/part-category.js:23]: why not do this instead 
```
const category = urlParams.get('category');
if(category) {
 return category;
 }
 return null; 
```
 if would take care the checking of empty string

**Review by Lakshmishri — CHANGES_REQUESTED** [review]

**Review by nfarmache-nc — APPROVED** [review]

**Review by Lakshmishri — APPROVED** [review]

**Review by manuel-vara — APPROVED** [review]

**Pull request merged**